### PR TITLE
Update node syncing content [Fixes #12003]

### LINKED ISSUE #12003: Various sync strategies are listed but not explained
In [the Nodes and Clients section of Developer Documents](https://ethereum.org/developers/docs/nodes-and-clients) There is a table comparing different execution clients under [Execution Clients](https://ethereum.org/developers/docs/nodes-and-clients#execution-clients). Until that point, the only part mentioned about sync strategies is "There are also options of different sync strategies which enable faster synchronization time. Synchronization refers to how quickly it can get the most up-to-date information on Ethereum's state." in [the Node Types](https://ethereum.org/developers/docs/nodes-and-clients#node-types)
In the table mentioned, there is a column for Sync Strategies, and these types are listed: Snap, Snap (without serving), Fast, Full. 
It would be helpful to define what those types mean

### PR DESCRIPTION
**Pull request by m4sterbunny**
## Description

Execution client summary table did not link users to sync types
Sync types were poorly identified and incomplete

Small proofs including:
- proof-of-stake > Proof of Stake (PoS) acronym then applied
- mainnet > Mainnet for Ethereum

## Issue
Fixes #12003

### CODE REVIEW

**Comment by m4sterbunny** [comment]
@wackerow I have tried to clarify and apply the common terms BUT it may be easier for users if we actually favor the term Checkpoint sync over the term Snap. Conceptually a Snap is the same as a Checkpoint sync for Consensys Layer, not sure having a different term for CL vs EL is actually adding benefit.
IF Besu team agree to move their custom "Checkpoint" sync to something like "Custom sync" -- then I believe we have a clear playing field to improve clarity by using a functional name. WDYT?

File: public/content/developers/docs/nodes-and-clients/index.md [hunk 1]
```diff
@@ -259,28 +276,26 @@ Light client mode downloads all block headers, block data, and verifies some ran
 
 **NB** Light sync does not yet work with proof-of-stake Ethereum - new versions of light sync should ship soon!
 
-[More on light clients](/developers/docs/nodes-and-clients/light-clients/)
+[More on light clients](/developers/docs/nodes-and-clients/light-clients/).
```
Comment (wackerow) [inline public/content/developers/docs/nodes-and-clients/index.md:279]: This is a convention used commonly on the site, without a trailing hard stop; please revert these

```suggestion
[More on light clients](/developers/docs/nodes-and-clients/light-clients/)
```

File: public/content/developers/docs/nodes-and-clients/index.md [hunk 2]
```diff
@@ -259,28 +276,26 @@ Light client mode downloads all block headers, block data, and verifies some ran
...
 
 ### Consensus layer sync modes {#consensus-layer-sync-modes}
 
 #### Optimistic sync {#optimistic-sync}
 
 Optimistic sync is a post-merge synchronization strategy designed to be opt-in and backwards compatible, allowing execution nodes to sync via established methods. The execution engine can _optimistically_ import beacon blocks without fully verifying them, find the latest head, and then start syncing the chain with the above methods. Then, after the execution client has caught up, it will inform the consensus client of the validity of the transactions in the Beacon Chain.
 
-[More on optimistic sync](https://github.com/ethereum/consensus-specs/blob/dev/sync/optimistic.md)
+[More on optimistic syncs](https://github.com/ethereum/consensus-specs/blob/dev/sync/optimistic.md).
```
Comment (wackerow) [inline public/content/developers/docs/nodes-and-clients/index.md:287]: It's singular in the other uses, I would leave this
```suggestion
[More on optimistic sync](https://github.com/ethereum/consensus-specs/blob/dev/sync/optimistic.md)
```

File: public/content/developers/docs/nodes-and-clients/index.md [hunk 3]
```diff
@@ -234,21 +234,38 @@ Synchronization modes represent different approaches to this process with variou
 
 ### Execution layer sync modes {#execution-layer-sync-modes}
 
-#### Full archive sync {#full-sync}
+#### Full node syncs
```
Comment (wackerow) [inline public/content/developers/docs/nodes-and-clients/index.md:237]:
```suggestion
#### Full node syncs {#full-node-syncs}
```

File: public/content/developers/docs/nodes-and-clients/index.md [hunk 4]
```diff
@@ -259,28 +276,26 @@ Light client mode downloads all block headers, block data, and verifies some ran
...
+A checkpoint sync, also known as weak subjectivity sync, creates a superior user experience for syncing a Beacon Node. It's based on assumptions of [weak subjectivity](/developers/docs/consensus-mechanisms/pos/weak-subjectivity/) which enables syncing the Beacon Chain from a recent weak subjectivity checkpoint instead of genesis. Checkpoint syncs make the initial sync time significantly faster with similar trust assumptions as syncing from [genesis](/glossary/#genesis-block).
 
-In practice, this means your node connects to a remote service to download recent finalized states and continues verifying data from that point. Third party providing the data is trusted and should be picked carefully.
+In practice, this means your node connects to a remote service to download recent finalized states and continues verifying data from that point. The third party providing the data is trusted and should be picked carefully.
 
-More on [checkpoint sync](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice)
+More on [checkpoint syncs](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice).
```
Comment (wackerow) [inline public/content/developers/docs/nodes-and-clients/index.md:295]: I'm not sure I'm a fan of changing to "syncs" here, but not a hill I'll die on

File: public/content/developers/docs/nodes-and-clients/index.md [hunk 5]
```diff
@@ -234,21 +234,38 @@ Synchronization modes represent different approaches to this process with variou
...
+
+##### Fast sync {#fast-sync}
 
-Snap sync verifies the chain block-by-block, just like a full archive sync; however, instead of starting at the genesis block, it starts at a more recent 'trusted' checkpoint that is known to be part of the true blockchain. The node saves periodic checkpoints while deleting data older than a certain age. Those snapshots are used to regenerate state data when it is needed, rather than having to store it all forever.
+Like a full sync, a fast sync downloads all blocks (including headers, transactions, and receipts). However, instead of re-processing the historical transactions, a fast sync relies on the receipts until it reaches a recent head, when it switches to importing and processing blocks to provide a full node. 
 
-- Fastest sync strategy, currently default in Ethereum mainnet
-- Saves a lot of disk usage and network bandwidth without sacrificing security
+- Fast sync strategy.
+- Reduces processing demand in favour of bandwith usage.
```
Comment (wackerow) [inline public/content/developers/docs/nodes-and-clients/index.md:259]:
```suggestion
- Reduces processing demand in favor of bandwidth usage.
```

**Review by wackerow — CHANGES_REQUESTED** [review]
Thanks @m4sterbunny! Left a few comments/requests to clean things up then think we can get this in soon

**Comment by wackerow** [comment]
Quick housekeeping notes... 

"12003(docs):fixes as per issue"

Issue Names: I see you're referencing an issue (12003). By convention, we'll place `[Fixes #12003]` at the end of the issue name. Please try to start with a brief summary title for your changes. Ie. "Update node syncing content [Fixes #12003]` ("fixes as per issue" isn't very helpful when skimming through issue names 😆 )

[More on auto-closing stuff](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

image

Issue Body: Try to use the default template to get started. Under a `## Description` you can describe in more detail your changes, then under `## Relate issue` you can add `- Fixes #12003` to cleanly link the PR to the issue it's fixing. 

Thanks!

File: public/content/developers/docs/nodes-and-clients/index.md [hunk 6]
```diff
@@ -234,21 +234,38 @@ Synchronization modes represent different approaches to this process with variou
...
 
-#### Full archive sync {#full-sync}
+#### Full node syncs
 
-Full sync downloads all blocks (including headers, transactions, and receipts) and generates the state of the blockchain incrementally by executing every block from genesis.
+You may run a full node with a:
+
+- [full sync](#full-sync)
+- [fast sync](#fast-sync)
+- [snap sync](#snap-sync)
```
Comment (wackerow) [inline public/content/developers/docs/nodes-and-clients/index.md:243]: I'm not sure having a bulleted list with hash links is necessary here... the content isn't very long, might suggest we just jump right into the sync options.

Also, ideally we keep the content tree structure to no deeper than h4 (`####`)... any way we could adjust this so we get straight into the options under "Execution layer sync modes"? like `#### Full node: full sync`, `#### Full node: fast sync`, `#### Full node: snap sync`?

File: public/content/developers/docs/nodes-and-clients/index.md (hunk 4 above)
Comment (m4sterbunny) [reply public/content/developers/docs/nodes-and-clients/index.md:295]: don't have a rationale other than "it sounds better out loud"!

File: public/content/developers/docs/nodes-and-clients/index.md (hunk 6 above)
Comment (m4sterbunny) [reply public/content/developers/docs/nodes-and-clients/index.md]: fixed with sentence structure instead

**Comment by m4sterbunny** [comment]
I think I resolved all @wackerow PTAL 
Apologies for poor hygiene, been lone writer for too long >> will learn to play nicely with others again

File: public/content/developers/docs/nodes-and-clients/index.md (hunk 6 above)
Comment (m4sterbunny) [reply public/content/developers/docs/nodes-and-clients/index.md]: fixed

File: public/content/developers/docs/nodes-and-clients/index.md [hunk 7]
```diff
@@ -271,16 +282,14 @@ Optimistic sync is a post-merge synchronization strategy designed to be opt-in a
...
+A checkpoint sync, also known as weak subjectivity sync, creates a superior user experience for syncing a Beacon Node. It's based on assumptions of [weak subjectivity](/developers/docs/consensus-mechanisms/pos/weak-subjectivity/) which enables syncing the Beacon Chain from a recent weak subjectivity checkpoint instead of genesis. Checkpoint syncs make the initial sync time significantly faster with similar trust assumptions as syncing from [genesis](/glossary/#genesis-block).
 
-In practice, this means your node connects to a remote service to download recent finalized states and continues verifying data from that point. Third party providing the data is trusted and should be picked carefully.
+In practice, this means your node connects to a remote service to download recent finalized states and continues verifying data from that point. The third party providing the data is trusted and should be picked carefully.
 
-More on [checkpoint sync](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice)
+More on [checkpoint syncs](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice).
```
Comment (wackerow) [inline public/content/developers/docs/nodes-and-clients/index.md:289]: @m4sterbunny Hah fair enough... I tend to lean towards consistency with these things... we're using "sync" in all of the headers and "More on snap sync", "More on optimistic sync" for the trailing read more links above... I'd suggest we just go with "More on checkpoint sync"

```suggestion
More on [checkpoint sync](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice)
```

Or if you disagree, please at least remove the trailing hard stop:

```suggestion
More on [checkpoint syncs](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice)
```

**Review by wackerow — APPROVED** [review]
Looks great, thanks a lot @m4sterbunny! Bringing in

**Comment by wackerow** [comment]
@all-contributors please add @m4sterbunny for content

**Pull request merged**